core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -804,6 +813,11 @@ oppia.controller('StateResponses', [
       responsesService.updateActiveAnswerGroup({
         rules: updatedRules
       });
+      var currentStateName = editorContextService.getActiveStateName();
+      StateSolutionHelperService.verifySolution(
```

Comment:
If verification fails there will be a flag on the solution editor which will say the current solution does not lead to a new state. Also, the exploration can be saved but not published unless this is corrected.